pycbc/waveform/bank.py

```diff
@@ -824,13 +824,21 @@ def __getitem__(self, index):
         # If available, record the total duration (which may
         # include ringdown) and the duration up to merger since they will be
         # erased by the type conversion below.
-        ttotal = template_duration = None
         if hasattr(htilde, 'length_in_time'):
             ttotal = htilde.length_in_time
-        if hasattr(htilde, 'chirp_length'):
+        else:
+            ttotal = None
+        # First try to load template_duration from get_waveform_filter,
```

Comment:
Shouldn't the template bank file have priority if it is there?

Reply:
As currently written - as a fallback - this is a simple addition. With Alex's proposal - as a change to the default behaviour - this becomes a potential problem. The reason is that some template bank codes populate a "template duration", which in many cases is different from what the search code is going to use (often around choice of f_lower). For safety, where PyCBC can compute the duration, it should do so ... I guess the alternative is that the bank2hdf code also becomes a "check the bank validity" code, and should check/update these values if needed.

Reply:
@spxiwh That's a good point. I think we should at least have a logging message though so that one can verify what actually happened. I know at some point this is going to cause someone a debugging headache.